datalad/tests/test_classes.py

```diff
@@ -0,0 +1,56 @@
+"""
+Tests to test implementation of basic classes for datalad.
+
+Note: There's not a lot to test by now.
+"""
+__author__ = 'Benjamin Poldrack'
+
+import os.path
+from shutil import rmtree
+
+from nose.tools import assert_raises, assert_is_instance, assert_true
+from git.exc import GitCommandError
+
+from datalad.support.gitrepo import GitRepo
+from datalad.support.annexrepo import AnnexRepo
+from datalad.support.dataset import Dataset
+
+def test_GitRepo():
+
+    pathToTestRepo = os.path.expanduser('~/test_git_repo')
+    gr = GitRepo(pathToTestRepo, 'http://psydata.ovgu.de/forrest_gump/.git')
+    assert_is_instance(gr, GitRepo, "GitRepo was not created.")
+    assert_true(os.path.exists(os.path.join(pathToTestRepo, '.git')))
+
+
+    #do it again should raise GitCommandError since git will notice there's already a git-repo at that path
+    with assert_raises(GitCommandError):
+        GitRepo(pathToTestRepo, 'http://psydata.ovgu.de/forrest_gump/.git')
```

Comment:
Agree. Was testing this, since there was a problem with that line at first
and just didn't change back afterwards.

2015-02-23 17:24 GMT+01:00 Yaroslav Halchenko <notifications@github.com>:

> In datalad/tests/test_classes.py
> <https://github.com/datalad/datalad/pull/42#discussion_r25177109>:
>
> > +
> > +from datalad.support.gitrepo import GitRepo
> > +from datalad.support.annexrepo import AnnexRepo
> > +from datalad.support.dataset import Dataset
> > +
> > +def test_GitRepo():
> > +
> > +    pathToTestRepo = os.path.expanduser('~/test_git_repo')
> > +    gr = GitRepo(pathToTestRepo, 'http://psydata.ovgu.de/forrest_gump/.git')
> > +    assert_is_instance(gr, GitRepo, "GitRepo was not created.")
> > +    assert_true(os.path.exists(os.path.join(pathToTestRepo, '.git')))
> > +
> > +
> > +    #do it again should raise GitCommandError since git will notice there's already a git-repo at that path
> > +    with assert_raises(GitCommandError):
> > +        GitRepo(pathToTestRepo, 'http://psydata.ovgu.de/forrest_gump/.git')
>
> heh -- never saw such use... I would have done
> assert_raises(GitCommandError, GitRepo, pathToTestRepo, 'url')
> for consistency
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/datalad/datalad/pull/42/files#r25177109>.
>